src/mqtt_client/mod.rs

```diff
@@ -96,21 +96,20 @@ enum Command<'a> {
 impl<'a> Command<'a> {
     fn from_message(topic: &'a str, value: &'a [u8]) -> Result<Self, ()> {
         let path = topic.strip_prefix('/').unwrap_or(topic);
-        let parsed = path
-            .split_once('/')
-            .map(|(head, tail)| (head, Some(tail)))
-            .unwrap_or((path, None));
-
-        match parsed {
-            ("list", None) => Ok(Command::List),
-            ("settings", Some(path)) => {
-                if value.is_empty() {
-                    Ok(Command::Get { path })
-                } else {
-                    Ok(Command::Set { path, value })
+
+        if path == "list" {
+            Ok(Command::List)
+        } else {
+            match path.split_once('/') {
+                Some(("settings", path)) => {
```

Comment:
Does this properly check that `path` is not None?

Reply:
In this arm `path` is always `&str`.